src/items.rs

```diff
@@ -1132,6 +1133,56 @@ fn format_struct(
     }
 }
 
+fn rewrite_bounds(
+    result: &mut String,
+    context: &RewriteContext<'_>,
+    bounds: &ast::GenericBounds,
+    terminator: &str,
+    shape: Shape,
+    span: Span,
+) -> Option<()> {
+    let indented = shape.block_indent(context.config.tab_spaces());
+    let items = itemize_list(
+        context.snippet_provider,
+        bounds.iter(),
+        terminator,
+        "+",
+        |bound| bound.span().lo(),
+        |bound| bound.span().hi(),
+        |bound| bound.rewrite(context, indented),
+        span.lo(),
+        span.hi(),
+        false,
+    )
+    .collect::<Vec<_>>();
+
+    let tactic = definitive_tactic(
+        &items,
+        ListTactic::LimitedHorizontalVertical(shape.width),
+        Separator::Plus,
+        context.config.max_width(),
+    );
+
+    let fmt = ListFormatting::new(indented, context.config)
+        .tactic(tactic)
+        .trailing_separator(SeparatorTactic::Never)
+        .separator("+")
+        .separator_place(SeparatorPlace::Front)
+        .align_comments(false);
+
+    let item_str = write_list(items, &fmt)?;
+
+    let space = if tactic == DefinitiveListTactic::Horizontal {
+        Cow::from(" ")
+    } else {
+        indented.indent.to_string_with_newline(&context.config)
+    };
+    result.push(':');
```

Comment:
Should we apply [`space_before_colon`](https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#space_before_colon)?